src/libs/security-hub-lib.ts

```diff
@@ -0,0 +1,131 @@
+import { IAMClient, ListAccountAliasesCommand } from "@aws-sdk/client-iam";
+import {
+  SecurityHubClient,
+  GetFindingsCommand,
+  GetFindingsCommandOutput,
+  Remediation,
+  AwsSecurityFinding,
+} from "@aws-sdk/client-securityhub";
+
+export interface SecurityHubFinding {
+  title?: string;
+  region?: string;
+  accountAlias?: string;
+  awsAccountId?: string;
+  severity?: string;
+  description?: string;
+  standardsControlArn?: string;
+  remediation?: Remediation;
+}
+
+export class SecurityHub {
+  private readonly region: string;
+  private readonly severityLabels: { Comparison: string; Value: string }[];
+  private accountAlias = "";
+
+  constructor({
+    region = "us-east-1",
+    severities = ["HIGH", "CRITICAL"],
+  } = {}) {
+    this.region = region;
+    this.severityLabels = severities.map((severity) => ({
+      Comparison: "EQUALS",
+      Value: severity,
+    }));
+    this.getAccountAlias().catch((error) => console.error(error));
+  }
+
+  private async getAccountAlias(): Promise<void> {
+    const iamClient = new IAMClient({ region: this.region });
+    const response = await iamClient.send(new ListAccountAliasesCommand({}));
+    this.accountAlias = response.AccountAliases?.[0] || "";
+  }
+
+  async getAllActiveFindings() {
+    try {
+      const securityHubClient = new SecurityHubClient({ region: this.region });
+
+      const currentTime = new Date();
+
+      // delay for filtering out ephemeral issues
+      const delayForNewIssues =
+        typeof process.env.SECURITY_HUB_NEW_ISSUE_DELAY !== "undefined"
```

Comment:
Is this just checking if `process.env. SECURITY_HUB_NEW_ISSUE_DELAY` is set? If so can you just do `!process.env. SECURITY_HUB_NEW_ISSUE_DELAY ? ...` rather than use `typeof`?

Reply:
I just copied this code over from the macpro code base. That does appear to be what they are doing. copilot suggests
const delayForNewIssues = +(process.env.SECURITY_HUB_NEW_ISSUE_DELAY ?? "86400000"); // 24 * 60 * 60 * 1000
const maxDatetime = new Date(currentTime.getTime() - delayForNewIssues);
Which looks good to me. The conversion from string to number creates some subtle problems which this handles reasonably well in 1 line of code.

Reply:
Did you want to add this change in for this PR?

Reply:
Added

Reply:
added